### PR TITLE
*: rename `PushDownFilter` to `FilterConditions`

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
Signed-off-by: Lloyd-Pottiger <yan1579196623@gmail.com>

### What problem does this PR solve?

Issue Number: ref #5829 

Problem Summary:

### What is changed and how it works?

Since the `PushDownFilter` will not pushed down to tablescan, the name will be misleading and conflict with filter which is really pushed down when late materialization is enable. So, rename `PushDownFilter` to `FilterConditions`

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Coprocessor/FilterConditions.h [hunk 1]
```diff
@@ -21,15 +21,22 @@
 
 namespace DB
 {
-struct PushDownFilter
+struct FilterConditions;
+using FilterConditionsPtr = std::shared_ptr<FilterConditions>;
```
Comment (SeaRise) [inline dbms/src/Flash/Coprocessor/FilterConditions.h:25]: why use `shared_ptr` here?

File: dbms/src/Flash/Planner/plans/PhysicalMockTableScan.cpp [hunk 2]
```diff
@@ -121,29 +121,29 @@ void PhysicalMockTableScan::updateStreams(Context & context)
 {
     mock_streams.clear();
     assert(context.mockStorage()->tableExistsForDeltaMerge(table_id));
-    mock_streams.emplace_back(context.mockStorage()->getStreamFromDeltaMerge(context, table_id, &push_down_filter));
+    mock_streams.emplace_back(context.mockStorage()->getStreamFromDeltaMerge(context, table_id, filter_conditions));
 }
 
-bool PhysicalMockTableScan::pushDownFilter(Context & context, const String & filter_executor_id, const tipb::Selection & selection)
+bool PhysicalMockTableScan::filterConditions(Context & context, const String & filter_executor_id, const tipb::Selection & selection)
```
Comment (SeaRise) [inline dbms/src/Flash/Planner/plans/PhysicalMockTableScan.cpp:127]: maybe rename to `setFilterConditions`?

Comment (Lloyd-Pottiger) [reply dbms/src/Flash/Planner/plans/PhysicalMockTableScan.cpp]: yep, changed.

File: dbms/src/Flash/Coprocessor/FilterConditions.h (hunk 1 above)
Comment (Lloyd-Pottiger) [reply dbms/src/Flash/Coprocessor/FilterConditions.h]: overthinking, reverted.

**Review by SeaRise — APPROVED** [review]
lgtm

**Comment by Lloyd-Pottiger** [comment]
unstable ut `ComputeServerRunner.runAggTasks`
https://ci.pingcap.net/blue/organizations/jenkins/tiflash-ghpr-unit-tests/detail/tiflash-ghpr-unit-tests/7765/pipeline cc @ywqzzy

File: dbms/src/Flash/Coprocessor/FilterConditions.h [hunk 3]
```diff
@@ -21,15 +21,20 @@
 
 namespace DB
 {
-struct PushDownFilter
+
+/** This struct FilterConditions is used to store the filter conditions of the selection whose child is a table scan.
+  * Those conditions will be used to construct roght index in storage engine.
```
Comment (ywqzzy) [inline dbms/src/Flash/Coprocessor/FilterConditions.h:26]:
```suggestion
  * Those conditions will be used to construct rough index in storage engine.
```

**Review by ywqzzy — APPROVED** [review]

**Review by ywqzzy — APPROVED** [review]

**Review by hongyunyan — APPROVED** [review]
LGTM

**Pull request merged**